src/local_scheduler/local_scheduler.cc

```diff
@@ -509,14 +513,35 @@ void reconstruct_task_update_callback(Task *task, void *user_context) {
   }
 }
 
+void log_put_reconstruction_error(Task *task, void *user_context) {
+  if (task == NULL) {
+    LOG_WARN(
+        "Reconstruction requested for a driver ray.put, unable to push error "
+        "to driver");
+    return;
+  }
+  LocalSchedulerState *state = (LocalSchedulerState *) user_context;
+  TaskSpec *spec = Task_task_spec(task);
+  FunctionID function = TaskSpec_function(spec);
+  push_error(state->db, TaskSpec_driver_id(spec),
+             PUT_RECONSTRUCTION_ERROR_INDEX, sizeof(function), function.id);
+}
+
 void reconstruct_evicted_result_lookup_callback(ObjectID reconstruct_object_id,
                                                 TaskID task_id,
+                                                bool is_put,
                                                 void *user_context) {
   /* TODO(swang): The following check will fail if an object was created by a
    * put. */
   CHECKM(!IS_NIL_ID(task_id),
          "No task information found for object during reconstruction");
   LocalSchedulerState *state = (LocalSchedulerState *) user_context;
+  if (is_put) {
+    task_table_get_task(state->db, task_id, NULL, log_put_reconstruction_error,
```

Comment:
Hmm yeah, I suspect that it will pretty much always be the case that we can't reconstruct a `ray.put` object, but I can change this so that we wait to see if the `task_table_test_and_update` was successful.